.github/workflows/tests.yml

```diff
@@ -39,7 +25,36 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  # detect whether any code changes are included in this PR
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      code: ${{ steps.filter.outputs.code }}
+    steps:
+      - uses: actions/checkout@v3
+        # dorny/paths-filter needs git clone for push events
+        # https://github.com/marketplace/actions/paths-changes-filter#supported-workflows
+        if: github.event_name == 'push'
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            code:
+              - 'conda/**'
+              - 'conda_env/**'
+              - 'tests/**'
+              - '*.py'
```

Comment:
There is Python code in the `docs` folder FWIW. Probably not a huge deal, but just wanted to mention it.